src/MergeContext.js

```diff
@@ -454,6 +454,214 @@ class BranchPosition
+    }
+
+    // basic checks for an unparsed message
+    _checkRaw(message) {
+        // CRs in CRLF sequences already removed
+        // other CRs are not treated specially (and are banned)
+        const trimmedMessage = this._trim(message);
+        const lines = trimmedMessage.split('\n');
+        for (let i = 0; i < lines.length; ++i) {
+            const line = lines[i];
+            const invalidPosition = this._invalidCharacterPosition(line);
+
+            if (invalidPosition !== -1)
+                throw new Error(`bad character at line ${i}, offset ${invalidPosition}`);
+
+            if (line.length > 72)
+                throw new Error(`too long line '${line}'`);
+        }
+    }
+
+    // parses attributes of the raw PR description
+    _parse(prDescription) {
+        const prDescriptionWithoutHeader = this._extractHeader(prDescription);
+
+        try {
+            const prDescriptionWithoutHeaderAndTrailer = this._extractTrailer(prDescriptionWithoutHeader);
+            this._parseBody(prDescriptionWithoutHeaderAndTrailer);
+        } catch (e) {
+            // TODO: supply debugString() prefix
+            Log.Logger.info("assuming no trailer: " + e.message);
+            this._parseBody(prDescriptionWithoutHeader);
+        }
+
+        assert(this._body !== undefined && this._body !== null);
+    }
+
+    // author is a {name, value}
+    _parseAuthor(author) {
+        const cred = author.value.match(/^ ([\w][^@<>,]*) <(\S+@\S+\.\S+)>$/);
+        if (!cred)
+            throw new Error(`invalid '${author.name}' value format`);
+
+        if (cred[0].includes(','))
+            throw new Error(`invalid '${author.name}' value: commas are not allowed`);
+
+        return {name: cred[1].trim(), email: cred[2].trim()};
+    }
+
+    _extractHeader(prDescriptionRaw) {
+        const prDescription = this._trim(prDescriptionRaw);
+        if (prDescription.startsWith('Authored-by:')) {
+            let tokenizer = new FieldsTokenizer(prDescription);
+            const authorField = tokenizer.nextField();
+            this._author = this._parseAuthor(authorField);
+            if (!tokenizer.atEnd())
+                throw new Error(`header must have nothing but a single 'Authored-by' attribute`);
+            return this._trim(tokenizer.remaining());
+        } else {
+            return prDescription;
+        }
+    }
+
+    _parseTrailer(trailerRaw) {
+        const trailer = this._trim(trailerRaw);
+        let tokenizer = new FieldsTokenizer(trailer);
+
+        if (tokenizer.atEnd())
+            throw new Error(`an empty trailer`);
+
+        while (!tokenizer.atEnd()) {
+            const field = tokenizer.nextField();
+            if (field.name === "Co-authored-by") {
+                this._parseAuthor(field);
+                const coAuthor = JSON.stringify(this._parseAuthor(field));
+                this._log(`accepting trailer field: ${coAuthor}`);
+            }
+            // and skip any other field
```

Comment:
We are missing a _checkForTypos() call for the skipped field. We must catch and ban Authored-by lines in the trailer! It may be a good idea to add a .raw member to the nextField() result tuple so that we can pass _that_ original string to _checkForTypos() instead of reconstructing its approximation here...